includes/acl/clip_reader/clip_reader.h

```diff
@@ -164,65 +145,57 @@ namespace acl
 			return false;
 		}
 
-		bool read_skeleton()
+		bool create_skeleton(std::unique_ptr<RigidSkeleton, Deleter<RigidSkeleton>>& skeleton)
 		{
 			SJSONParser::State before_bones = m_parser.save_state();
 
 			uint16_t num_bones;
 			if (!process_each_bone(nullptr, num_bones))
-			{
 				return false;
-			}
 
 			m_parser.restore_state(before_bones);
 
-			std::unique_ptr<RigidBone, Deleter<RigidBone>> bones = allocate_unique_type_array<RigidBone>(m_allocator, num_bones);
+			std::unique_ptr<RigidBone, Deleter<RigidBone>> bones = make_unique_array<RigidBone>(m_allocator, num_bones);
 			if (!process_each_bone(bones.get(), num_bones))
-			{
 				return false;
-			}
 
-			m_skeleton = allocate_shared_type<RigidSkeleton>(m_allocator, m_allocator, bones.get(), num_bones);
+			skeleton = make_unique<RigidSkeleton>(m_allocator, m_allocator, bones.get(), num_bones);
 
 			return true;
 		}
 
+		bool read_skeleton()
+		{
+			uint16_t num_bones;
+			return process_each_bone(nullptr, num_bones);
+		}
+
 		bool process_each_bone(RigidBone* bones, uint16_t& num_bones)
```

Comment:
I usually prefix output only arguments with out_, e.g. out_num_bones but not always, I struggle to be consistent at times